tests/framework/installer/install_helper.go

```diff
@@ -330,3 +335,27 @@ func NewK8sRookhelper(clientset *kubernetes.Clientset, t func() *testing.T) *Ins
 		T:           t,
 	}
 }
+
+func IsAdditionalDeviceAvailableOnCluster() bool {
+	executor := &exec.CommandExecutor{}
+	devices, err := sys.ListDevices(executor)
```

Comment:
what about using the `DiscoverDevices` function instead? it takes care of filtering out disk types you don't want and getting the properties for you.  That function may make what this function is trying to do easier.
https://github.com/rook/rook/blob/master/pkg/clusterd/disk.go#L96

Note this isn't taking into account child partitions or filesystems either.  You should consider using the `checkIfDeviceAvailable` function as well: https://github.com/rook/rook/blob/master/pkg/ceph/osd/device.go#L122

Reply:
@jbw976  the `DiscoverDevices` function is returning all the partitions in a disk(local disks), if the disk has no partitions then nothing is returned eg
```2017-10-26 22:21:28.517165 W | inventory: skipping device xvda with an unknown uuid. Failed to complete get disk xvda uuid: exit status 2
2017-10-26 22:21:28.517183 I | exec: Running command: lsblk /dev/xvda1 --bytes --nodeps --pairs --output SIZE,ROTA,RO,TYPE,PKNAME
2017-10-26 22:21:28.518227 I | exec: Running command: df --output=source,fstype
2017-10-26 22:21:28.520796 I | exec: Running command: lsblk /dev/xvdb --bytes --nodeps --pairs --output SIZE,ROTA,RO,TYPE,PKNAME
2017-10-26 22:21:28.521811 I | exec: Running command: sgdisk --print /dev/xvdb
2017-10-26 22:21:28.523287 W | inventory: skipping device xvdb with an unknown uuid. Failed to complete get disk xvdb uuid: exit status 2
2017-10-26 22:21:28.523315 I | installer: cluster devices : [0xc420302ea0]```
so `DiscoverDevices` is not finding my second disk `xvdb` which i want to use

However once it use the  the disk and and create a block device, i see partitions on the 2nd disk. And the partitions in the 2nd disk donve have a fileSystem(so they are raw devices). 
eg ```2017-10-26 22:23:20.781104 I | installer: cluster devices : [0xc420303830 0xc42008e2d0 0xc42008e360 0xc420302a20]
2017-10-26 22:23:20.781119 I | installer: available devices : %!(EXTRA []string=[])
2017-10-26 22:23:20.781127 I | installer: name: xvda1,type: part,children: %!s(bool=false),parent: xvda,filesystem: ext4,Id: 
2017-10-26 22:23:20.781131 I | installer: name: xvdb1,type: part,children: %!s(bool=false),parent: xvdb,filesystem: ,Id: 
2017-10-26 22:23:20.781136 I | installer: name: xvdb2,type: part,children: %!s(bool=false),parent: xvdb,filesystem: ,Id: 
2017-10-26 22:23:20.781147 I | installer: name: xvdb3,type: part,children: %!s(bool=false),parent: xvdb,filesystem: ,Id: ```

And the fumction `checkIfDeviceAvailable` is not global so i cannot use it.

With the current changes, i am looking to see the cluster has more than one disk. If we find more that one disk we will use `useAllDevices: true` in not we use `useAllDevices: false`.
During our CI pipeline, we are provisioning a VM on AWS with a additional raw device. So when we run tests on AWS, we end up using `useAllDevices: true` option on the rook cluster and implicitely test raw device. 
Is the approach ok?

Reply:
Yes, it was my hope that you'd get to reuse code instead of having to reimplement existing logic in the integration tests that already exists elsewhere.

However, if `DiscoverDevices` skips over your extra raw device you added (`skipping device xvdb with an unknown uuid.`), then won't the OSDs fail to find and use it also when you run a test with `useAllDevices: true`? I wouldn't expect the cluster to come up functional with any OSDs using disks.

Reply:
We where using the 2nd disk. I  could tell becuase the space avaliable on the additional  disk went down  only after i started running the tests.  If there are no disks avaliable and I use `useAllDevices: true`, all the tests were failing. Also, Once the disk was used by a cluster, i couldn't reuse it. I was errors when in the multiCluster Test when i started both clusters with `useAllDevices: true`. The tests stated passing after I set `useAllDevices: true` on first cluster and `useAllDevices: 2nd` end cluster. 

Also after looking a the test log 
 `http://jenkins.rook.io/blue/rest/organizations/jenkins/pipelines/rook/pipelines/rook/branches/PR-1140/runs/16/nodes/44/steps/84/log/?start=0`,  it looks like for the first test the `xvdb`disk has not partitions, and from the 2nd test onwards you see partitions on it. Makes me believe the first test actually used the device even though `DiscoverDevices` skipped it while checking for partitions.

Reply:
I also see what you are seeing in the test log output, but it's hard to be 100% confident that the OSDs used and partitioned them correctly because we aren't looking at the details such as partition labels.

The `Failed to complete get disk xvdb uuid` error has exit status 2 from `sgdisk` which according to https://www.rodsbooks.com/gdisk/sgdisk.html means it failed to read the partition table of the disk. My only theory here is that the OSD pod runs privileged and perhaps the test code does not, so maybe the OSDs had no issues getting the device details while the test code was not allowed to.

Is it possible to get the OSD pod logs for these runs just so we can confirm 100% that they are using the devices OK?  Another way to do it would be to SSH to the slave after the tests are run and execute a `lsblk -o name,type,size,partlabel`.  I'd like to fully understand this before proceeding, thanks!

Reply:
OK now that we have confirmed in the pod logs that the OSD is using the xvdb device OK, I think your original approach is sufficient.  You can remove these calls to discoverDevices and GetAvailableDevices and just rely on counting the number of disks from the ListDevices output.  Then squash everything and I think we'll be good to go.  Thanks for the extra work to confirm the tests were behaving as intended.